org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java

```diff
@@ -214,6 +247,77 @@ public String writeCsar(IRepository repository, DefinitionsChildId entryId, Outp
+                ubuntuFlag = true;
+            }
+
+            if (tNodeTemplate.getDeploymentArtifacts() != null) {
+                @NonNull List<TDeploymentArtifact> deploymentArtifacts = tNodeTemplate.getDeploymentArtifacts().getDeploymentArtifact();
+
+                for (TDeploymentArtifact deploymentArtifact :
+                    deploymentArtifacts) {
+
+                    if (deploymentArtifact.getArtifactType().getLocalPart().equals("DockerContainerArtifact")) {
+                        dockerFlag = true;
+                    }
+                }
+            }
+
+            if (dockerFlag || ubuntuFlag) {
+                TNodeTemplate duplicatedNode = BackendUtils.clone(tNodeTemplate);
+                duplicatedNode.setId(tNodeTemplate.getName());
+                duplicatedNode.setName(tNodeTemplate.getName());
+
+                QName tNodeQtype = tNodeTemplate.getTypeAsQName();
+                QName qName = new QName(tNodeQtype.getNamespaceURI(), tNodeQtype.getLocalPart() + "-self", tNodeQtype.getPrefix());
+
+                duplicatedNode.setType(qName);
+
+                if (dockerFlag) {
+                    @Nullable QName artifactRef = duplicatedNode.getDeploymentArtifacts().getDeploymentArtifact().get(0).getArtifactRef();
+                    String artifactTemplateName = "{http://opentosca.org/artifacttemplates}" + artifactRef.getLocalPart() + "-self";
+
+                    ArtifactTemplateId newArtifactTemplateId = new ArtifactTemplateId(QName.valueOf(artifactTemplateName));
+
+                    TArtifactTemplate artifactTemplate = repository.getElement(newArtifactTemplateId);
+
+                    List<TDeploymentArtifact> artifactTemplateList = new ArrayList<>();
+                    artifactTemplateList.add(new TDeploymentArtifact.Builder(artifactTemplate.getName(), artifactTemplate.getType())
+                        .setArtifactRef(QName.valueOf("{http://opentosca.org/artifacttemplates}" + artifactTemplate.getId())).build());
+
+                    TDeploymentArtifacts DAs = new TDeploymentArtifacts.Builder(artifactTemplateList).build();
+
+                    duplicatedNode.setDeploymentArtifacts(DAs);
+                }
+
+                topologyTemplateCopy.getNodeTemplateOrRelationshipTemplate().add(duplicatedNode);
+                topologyTemplateCopy.getNodeTemplateOrRelationshipTemplate().remove(tNodeTemplate);
+                topologyTemplateCopy.setNodeTemplates(topologyTemplateCopy.getNodeTemplates());
+                List<TRelationshipTemplate> oldOutgoingRelationships = ModelUtilities.getOutgoingRelationshipTemplates(topologyTemplateCopy, tNodeTemplate);
+
+                List<TRelationshipTemplate> newOutgoingRelationships
+                    = resetOutgoingRelationships(oldOutgoingRelationships, duplicatedNode);
+                topologyTemplateCopy.getNodeTemplateOrRelationshipTemplate().removeAll(oldOutgoingRelationships);
```

Comment:
I guess that if you just update the source/target of the relationships, its enough, you don't have to create new ones.